www/js/app.js

```diff
@@ -111,11 +117,15 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     // Define behavior of HTML elements
     var searchArticlesFocused = false;
     $('#searchArticles').on('click', function() {
+        var prefix = document.getElementById('prefix').value;
+        // Do not initiate the same search if it is already in progress
+        if (globalstate.search.prefix === prefix && !/cancelled|complete/.test(globalstate.search.status)) return;
```

Comment:
It's minor, but the regexp should try to match the whole string. As it is currently, it can also match a substring, which would be wrong (even if it can not happen as the values are all known and set by us), and is probably slightly slower too.

Reply:
OK, no problem.